---
title: "RNAs-seq"
subtitle: "Counting on expression"

format:
  html:
    code-fold: true
    code-tools: FALSE
    code-copy: true
    highlight-style: github
    code-overflow: wrap
---

# Reading

- [Opportunities in Functional Genomics: A Primer on Lab and Computational Aspects](https://d.pr/f/ngmWkw)

- [Fitting Multifactorial Models of Differential Expression](https://marineomics.github.io/FUN_01_DGE_comparison_v2.html)


# Objectives
Learn how to take raw RNA sequence data and perform differential gene expression analysis.


# Sequence 


A FASTQ file has four line-separated fields per sequence:

- Field 1 begins with a '@' character and is followed by a sequence identifier and an optional description (like a FASTA title line).
- Field 2 is the raw sequence letters.
- Field 3 begins with a '+' character and is optionally followed by the same sequence identifier (and any description) again.
- Field 4 encodes the quality values for the sequence in Field 2, and must contain the same number of symbols as letters in the sequence.

https://en.wikipedia.org/wiki/FASTQ_format

See also page 341 of the textbook.




{{< video https://youtu.be/5LYnq84Pjzk width="400" height="300">}}





